packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -253,17 +273,36 @@ export function useAsyncList<T, C = string>(options: AsyncListOptions<T, C>): As
     let abortController = new AbortController();
     try {
       dispatch({...action, abortController});
+      let previousFilterText = action.filterText ?? data.filterText;
+
+      if (action.type !== 'success' && action.type !== 'update') {
+        loadingState.current = action.type;
+      } else {
+        loadingState.current = 'idle';
```

Comment:
opinions on this? Not sure a case exists where dispatchFetch is called w/ `action=success/update`. I could also modify dispatch in such a way that loadingState ref is modified in the reducer...

Reply:
Hmm I don't think a case exists where action.type would be success or update calling dispatchFetch.

Reply:
yeah, figured as much. Maybe I'll just delete the else part